wandb/sdk/internal/sender.py

```diff
@@ -996,7 +1034,7 @@ def send_run(self, record: "Record", file_dir: Optional[str] = None) -> None:
         else:
             logger.info("updated run: %s", self._run.run_id)
 
-    def _init_run(
+    def _init_run(  # noqa: C901
```

Comment:
we have the lint for a reason; either the original code should be refactored or the additions should be written such that we don't need to disable the "function is too complex" check

Reply:
i refactored it